Feature/gui sector model

RoaldL opened the pull request:
This pull request repairs the SectorModel configuration pages and forms and implements new components for the new Spec input/output/parameter metadata.

RoaldL commented:
Ready for review. Please note that a roundtrip read/save of a SectorModelConfig causes the dimension coordinates to be written to the sector model config. (see https://www.pivotaltracker.com/story/show/160617312)

willu47 commented:
Travis build is failing due to tests failing for `tests/http_api/test_crud.py`.  You can use pytest.mark.skip to temporarily skip the tests that you will fix up in later PRs.

RoaldL commented:
@willu47 
Spec Form
- https://www.pivotaltracker.com/story/show/160649408
Model Wrappers form
- There was a bug in the way the page was initialised. This is now fixed and the form should show the behavior as you described
- https://www.pivotaltracker.com/story/show/160651444

@tomalrussell 
- I have removed the scenario_set and narrative_set tests as they are not part of the datainterface anymore. I have added a test for the new dimensions http-api.
- This is done now, tests should pass now.
